src/components/templates/AnimateMap/AnimateMap.tsx

```diff
@@ -0,0 +1,85 @@
+import * as PIXI from "pixi.js";
+
+import React, { useState, useRef, useEffect } from "react";
+import { Provider, ReactReduxContext } from "react-redux";
+import { AppConsumer, Container, Stage, useApp } from "@inlet/react-pixi";
+import { AnimateMapVenue } from "../../../types/venues";
+import { useUser } from "hooks/useUser";
+import { useSelector } from "../../../hooks/useSelector";
+import { animateMapStageOptionsSelector } from "../../../utils/selectors";
+
+import "./AnimateMap.scss";
+
+import { MapContainer } from "./components/Map/MapContainer";
+import { MAP_IMAGE } from "./constants/Resources";
+
+export interface AnimateMapProps {
+  venue: AnimateMapVenue;
+}
+
+export const AnimateMap: React.FC<AnimateMapProps> = () => {
+  const options = useSelector(animateMapStageOptionsSelector);
+
+  const containerRef = useRef<HTMLDivElement>(null);
+  const { user, profile } = useUser();
+
+  if (!user || !profile) {
+    return <>Loading..</>;
+  }
+
+  const container = containerRef.current ?? null;
+
+  return (
+    <div ref={containerRef} className="animatemap-venue-container">
+      {container && (
+        <ReactReduxContext.Consumer>
+          {({ store }) => (
+            <Stage
+              width={container.offsetWidth}
+              height={container.offsetHeight}
+              options={Object.assign(options, { resizeTo: container })}
+            >
+              <AppConsumer>
+                {(app) => (
+                  <Provider store={store}>
+                    {app ? <LoadingContainer /> : <Container />}
+                  </Provider>
+                )}
+              </AppConsumer>
+            </Stage>
+          )}
+        </ReactReduxContext.Consumer>
+      )}
+    </div>
+  );
+};
+
+const LoadingContainer = () => {
+  const app = useApp();
+
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    if (!isLoading && app) {
+      setIsLoading(true);
+
+      app.loader.reset().add(MAP_IMAGE).load();
+      app.loader.onLoad.add(() => setIsLoaded(true));
+      app.loader.onError.add(
+        (loader: PIXI.Loader, resource: PIXI.LoaderResource) =>
+          console.error(resource)
+      );
+    }
+  }, [isLoading, app]);
```

Comment:
Instead of nesting entire blocks within an `if`, we use 'guard statements' at the start of the block, that 'shortcut' by returning when the condition isn't met. Note that there should always be a blank line of space after a 'guard statement' like this,=:

```suggestion
  useEffect(() => {
    if (isLoading || !app) return;

    setIsLoading(true);

    app.loader.reset().add(MAP_IMAGE).load();
    app.loader.onLoad.add(() => setIsLoaded(true));
    app.loader.onError.add(
      (loader: PIXI.Loader, resource: PIXI.LoaderResource) =>
        console.error(resource)
    );
  }, [isLoading, app]);
```

Reply:
Does `app.loader` need to be 'cleaned up' in the `useEffect`'s cleanup return?

- https://reactjs.org/docs/hooks-reference.html#cleaning-up-an-effect

Reply:
Usually we need a callback for "cleanup". But now we don't need it. Moreover, in one of the next sprints, we will one way or another take on a full-fledged resource management system and all this code will be rework.

Reply:
Fair, makes sense.